File: app/src/main/java/org/digitalcampus/oppia/gamification/GamificationBroadcastReceiver.java
package org.digitalcampus.oppia.gamification;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import org.digitalcampus.oppia.listener.GamificationEventListener;

public class GamificationBroadcastReceiver extends BroadcastReceiver {

    private GamificationEventListener listener;


    @Override
    public void onReceive(Context context, Intent intent) {

        String message = intent.getStringExtra(GamificationService.SERVICE_MESSAGE);
        int points = intent.getIntExtra(GamificationService.SERVICE_POINTS, 0);

        if (listener != null){
            listener.onEvent(message, points);
        }

        abortBroadcast();
    }

    public void setGamificationEventListener(GamificationEventListener listener){
        this.listener = listener;
    }
}

File: app/src/main/java/org/digitalcampus/oppia/gamification/GamificationBroadcastReceiver.kt
package org.digitalcampus.oppia.gamification

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.digitalcampus.oppia.listener.GamificationEventListener

class GamificationBroadcastReceiver : BroadcastReceiver() {
    private var listener: GamificationEventListener? = null

    override fun onReceive(context: Context, intent: Intent) {
        val message = intent.getStringExtra(GamificationService.SERVICE_MESSAGE)
        val points = intent.getIntExtra(GamificationService.SERVICE_POINTS, 0)
        listener?.onGamificationEvent(message, points)
        abortBroadcast()
    }

    fun setGamificationEventListener(listener: GamificationEventListener) {
        this.listener = listener
    }
}